web/src/message_list_view.js

```diff
@@ -986,15 +1030,9 @@ export class MessageListView {
         // than auto-scrolling.
         const started_scrolled_up = message_viewport.is_scrolled_up();
 
-        // The messages we are being asked to render are shared with between
-        // all messages lists. To prevent having both list views overwriting
-        // each other's data we will make a new message object to add data to
-        // for rendering.
-        const message_containers = messages.map((message) => {
+        for (const message of messages) {
             message.url = hash_util.by_conversation_and_time_url(message);
```

Comment:
That comment appears to have been wrong, since we don't do a proper copy here. I guess the `.url` field is modifying the shared copy each time, which basically works but is definitely evil.